docroot/themes/contrib/civictheme/includes/pager.inc

```diff
@@ -53,3 +56,82 @@ function civictheme_preprocess_pager(&$variables) {
   // Add CSS class for ajax pagination.
   $variables['pager_items_modifier_class'] = 'js-pager__items';
 }
+
+/**
+ * Implements template_preprocess_views_mini_pager().
+ *
+ * @SuppressWarnings(PHPMD.StaticAccess)
+ */
+function civictheme_preprocess_views_mini_pager(&$variables) {
+  /** @var \Drupal\Core\Pager\PagerManagerInterface $pager_manager */
+  $pager_manager = \Drupal::service('pager.manager');
+  if (isset($variables['items']['current'])) {
+    $variables['items']['pages'] = [
+      $variables['items']['current'] => [
+        'href' => '#',
+      ],
+    ];
+  }
+
+  $element = $variables['element'] ?? NULL;
+  if ($element === NULL) {
+    return;
+  }
+
+  $pager = $pager_manager->getPager($element);
+  if (!$pager) {
+    return;
+  }
+
+  $variables['total_pages'] = $pager->getTotalPages();
+  $variables['current'] = $variables['current'] ?? $pager->getCurrentPage() + 1;
+
+  $view = $variables['view'];
+  if ($view) {
+    unset($variables['view']);
+    _civictheme_preprocess_pager__items_per_page($variables, $view, $pager);
+  }
+
+  _civictheme_preprocess_mini_pager__ellipsis($variables);
+}
+
+/**
+ * Helper to preprocess items_per_page in pagination component.
+ */
+function _civictheme_preprocess_pager__items_per_page(&$variables, $view, $pager) {
+  /** @var \Drupal\views\ViewExecutable $view */
+  $view_pager = $view->getPager();
+  if (!empty($view_pager->options['expose']['items_per_page']) && $view_pager->options['expose']['items_per_page'] === TRUE) {
+    $options = $view_pager->options;
+    $variables['items_per_page_label'] = $options['expose']['items_per_page_label'];
+    $current_items_per_page = (int) $pager->getLimit();
+    unset($variables['attributes']['#view']);
+    $exposed_form = $view->display_handler->getPlugin('exposed_form');
+    $exposed_widgets = $exposed_form->renderExposedForm();
+    $items_per_page = [];
+    $items_per_page_options = explode(',', $view_pager->options['expose']['items_per_page_options']);
+    foreach ($items_per_page_options as $value) {
+      $items_per_page[] = [
+        'type' => 'option',
+        'label' => $value,
+        'value' => (int) $value,
+        'selected' => (int) $value === $current_items_per_page,
+      ];
+    }
+    $variables['items_per_page_options'] = $items_per_page;
+    $variables['items_per_page_attributes'] = 'name="items_per_page" form="' . $exposed_widgets['#id'] . '"';
```

Comment:
I know you have asked me to leave form components - but items_per_page is a form component. I have added the `form` attribute to the select which then links it to the view exposed form when submitting.